components/table/table-controls.js

```diff
@@ -17,6 +17,18 @@ class TableControls extends SelectionControls {
 		};
 	}
 
+	static get styles() {
+		return [super.styles, css`
+			:host {
+				--d2l-selection-controls-background-color: var(--d2l-table-controls-background-color);
+				z-index: 2; /* must be greater than d2l-table rows */
+			}
+			:host([no-sticky]) {
+				z-index: auto;
+			}
+		`];
+	}
```

Comment:
I called this out in the description, but this code is "sort of" a placeholder - it's safe to merge it as is and will improve the demos, or we can wait for #3283 to merge first. Note that in #3283 I ended up using `z-index: 4` instead of `z-index: 2`.